dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/AbstractAnalyticsService.java

```diff
@@ -544,12 +544,18 @@ private List<String> getDimensionItemUidList( EventQueryParams params, QueryItem
         {
             return item.getOptionSetFilterItemsOrAll();
         }
-        else
+        else if ( itemOptions != null )
         {
             return itemOptions.stream()
                 .map( BaseIdentifiableObject::getUid )
                 .collect( toList() );
         }
+
+        // equal to no option set and no legend set, derived from query
```

Comment:
I think this comment is not super clear, perhaps we could rewrite it to make it a bit clearer?

Reply:
Done